certbot-apache/certbot_apache/interfaces.py

```diff
@@ -250,6 +297,28 @@ class BlockNode(ParserNode):
     children: Tuple[ParserNode, ...]
     """
 
+    # pylint: disable=too-many-arguments, super-init-not-called
+    @abc.abstractmethod
+    def __init__(self, name="", parameters=(), children=(), ancestor=None,
```

Comment:
I think we should remove this `children` parameter and instead just use the `add_child*` methods. The reason for this is:

1. Proper children need to have `ancestor` set which is impossible before this object is created.
2. During parsing, I'm struggling to imagine a scenario where we've parsed a block's children before the block itself.